amoro-ams/src/test/java/org/apache/amoro/server/TestDefaultOptimizingService.java

```diff
@@ -205,15 +211,25 @@ public void testPollTaskThreeTimes() {
 
     // 4.retry poll task
     OptimizingTask task2 = optimizingService().pollTask(token, THREAD_ID);
-    Assertions.assertEquals(task2, task);
+    Assertions.assertEquals(task2.getTaskId(), task.getTaskId());
+    Assertions.assertNotEquals(task2.getTaskInput(), task.getTaskInput());
+    TableOptimizing.OptimizingInput input =
+        SerializationUtil.simpleDeserialize(task.getTaskInput());
+    TableOptimizing.OptimizingInput input2 =
+        SerializationUtil.simpleDeserialize(task2.getTaskInput());
+    Assertions.assertEquals(input2.toString(), input.toString());
```

Comment:
Make changes here because `RewriteFilesInput` is not equals after serialized via JDK9+. The `MixedTable` can not keep idential bytes